openedx/core/djangoapps/user_api/preferences/views.py

```diff
@@ -0,0 +1,147 @@
+"""
+NOTE: this API is WIP and has not yet been approved. Do not use this API without talking to Christina or Andy.
+
+For more information, see:
+https://openedx.atlassian.net/wiki/display/TNL/User+API
+"""
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework import permissions
+
+from openedx.core.lib.api.parsers import MergePatchParser
+from ..api.account import AccountUserNotFound, AccountNotAuthorized
+from ..models import PreferenceNotFound, PreferenceUpdateError
+from .api import (
+    get_user_preference, get_user_preferences, set_user_preference, update_user_preferences, delete_user_preference
+)
+
+
+class PreferencesView(APIView):
+    """
+        **Use Cases**
+
+            Get or update the user's preference information. Updates are only supported through merge patch.
+
+        **Example Requests**:
+
+            GET /api/user/v0/preferences/{username}/
+
+            PATCH /api/user/v0/preferences/{username}/ with content_type "application/merge-patch+json"
+
+        **Response Value for GET**
+
+            A dict will be returned with key/value pairs (all of type String).
+
+        **Response for PATCH**
+
+             Returns a 204 status if successful, with no additional content.
```

Comment:
I recommend adding more information about errors thrown (see what I documented for accounts).